Feature/p events

Div685 opened the pull request:
# Private Event

This is a Private-event project using ruby and rails. This contains the following sections:
- User signup and Sign In
- Create Events
-  Invite Users
- Accept Invitations

### Built With

- [Rails](https://guides.rubyonrails.org/)
- [Ruby](https://www.ruby-lang.org/en/)
- HTML
- Bootstrap
- SQLite3

Div685 committed:
Add feature-branch

- update readme

Div685 committed:
Add bootstrap

- update files for bootstrap setup

Div685 committed:
Add User model

- Migrated User model using rails db:migrate

Div685 committed:
Add User controller

- Add new, create, index, show methods

Div685 committed:
update routes for users

- Add root routes
- update user controller
- update application_controller
- Add navbar

Div685 committed:
Add Session controller

- Add new, create, destroy method in session controller
- Add html code to create login form
- Add routes to sessions

Div685 committed:
Add Login and sign_up part

- style both login and sign_up html
- update routes

Div685 committed:
Add show.html.erb

- update routes
- update user_controller

Div685 committed:
Add Event model

- Add association to user and event model

Div685 committed:
Add user_id to events

- Add foreign_key to event model

Div685 committed:
Add Event controller

- Add index, new, create methods
- Add event_params method
- Add routes

Div685 committed:
Add create event form

- Add new, show html file
- update events_controller.rb

Div685 committed:
update navbar

- Display user Name in navbar
- update show user html file

Div685 committed:
Add Attendance model

- Add association with model

Div685 committed:
Update user and event model

- add foreign_key, scope for date
- update routes
- update index, show.html page for attendance and events

Div685 committed:
Add style to index page and users page

- update event and user controller
- Add _past-events and _upcomming-events as file
- update routes
- update application.html.erb

Div685 committed:
Add status to attenances using migrations

Div685 committed:
update user_controller

- Add @upcoming and @past in show method

Div685 committed:
Add font-awesome

- using yarn to install font-awesome
- update show.html.erb event file

Div685 committed:
update user, event model

- update event_controller

Div685 committed:
Add attendances index page

- display list of users and invite button to index page

Div685 committed:
Add logic to helper class

- separated logic from view file

Div685 committed:
Add invitation to user part

- Add Withdraw, join and invite button

Div685 committed:
Add html style in view pages

- event show page
- user show page
- user index page

Adewunmi97 committed:
update attendances view file

Adewunmi97 committed:
add bootstrap styling to user index file

Adewunmi97 committed:
add styles to users.scss

Adewunmi97 committed:
add readme file contents

Div685 committed:
Add Screenshot

- update readme.md file for screenshot
- update user.scss in user index

Div685 committed:
update files for linters

- Used rubocop linter
- used stylelint linter

peterrobert commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,86 @@
+  def index
+    @users = User.where('id != ?', current_user.id).order('name ASC')
+    @event = Event.find(params[:event_id])
+    # @current_user = session['current_user']
```
Remove commented code. It will make your code clean

peterrobert reviewed (changes requested):
### CHANGES REQUIRED ♻️ 🛑 

Hello, @Div685 and @Adewunmi97 very nice work 👍🏿 👏🏿  your project looks very nice I love the design. 👏🏿  You have implemented almost all the requirements. Your project looks perfect. However, it seems you forgot one very important part of this project. Testing.  Kindly check below ⬇️ 

- [x] You need to create tests for the models for validations and associations  ⬇️ 
 
![unit_tests_requirement](https://user-images.githubusercontent.com/32361963/103763643-30f35000-502b-11eb-8228-ce15697b680b.png)

- [x] Your authentication system should have integration tests (Rspec + Capybara).
  
![integration_tests_requirement](https://user-images.githubusercontent.com/32361963/103763819-7021a100-502b-11eb-8097-7acf91f038d3.png)

- [x] event managment features have integration tests (Rspec + Capybara)
   
![integration_tests_requirement](https://user-images.githubusercontent.com/32361963/103763916-92b3ba00-502b-11eb-9ca6-4e35e56f8e4a.png)


### if you have any questions kindly reach me on slack @peterrobert

### HAPPY CODING 💻

Div685 committed:
Add and install rspec

- Add capybara and it's configuration
- update Gemfile

Div685 committed:
Add validation to models

- user, event models

Div685 committed:
Add spec files

- Add event, user, and attendance spec file

Div685 committed:
Add rspec controller

- remove comment from user controller

Div685 committed:
Add Event controller rspec

- Add create event test using rspec

Div685 committed:
update rspec files for linters

teekaytech reviewed (approved):
# Status: Approved :green_circle: 

### Congratulations team, this project has passed all the mandatory requirements. It is therefore approved :heavy_check_mark:

**OPTIONAL CHANGE(s)**
_Although the project is approved, please make the following changes before you merge._
- [x] There should be no logic in your view. All view-related logic should be moved to the helper module.
- You're having logic (if statements) in some of your views ( app/views/events/show.html.erb, app/views/users/index.html.erb, etc). Kindly move all view-related logic to their corresponding helper modules.


Again, congratulations, you can merge to master.
Good luck with your next task.

* * * 
**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

Div685 committed:
update files to separet logic from view

- update index user file
- update event show file
- add validation error to sign_up page

Div685 committed:
update file to correct linter
